Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -8414,98 +8352,88 @@ void CPhysicalGeometry::SetBoundTecPlot(char mesh_filename[MAX_STRING_SIZE], boo
-    nodes->SetColor(iPoint, 0);
+void CPhysicalGeometry::SetColorGrid_Parallel(const CConfig *config) {
 
   /*--- We need to have parallel support with MPI and have the ParMETIS
    library compiled and linked for parallel graph partitioning. ---*/
 
-#ifdef HAVE_MPI
-#ifdef HAVE_PARMETIS
+#if defined(HAVE_MPI) && defined(HAVE_PARMETIS)
+
+  /*--- Only call ParMETIS if we have more than one rank to avoid errors ---*/
+
+  if (size == SINGLE_NODE) return;
 
   MPI_Comm comm = MPI_COMM_WORLD;
 
   /*--- Linear partitioner object to help prepare parmetis data. ---*/
 
   CLinearPartitioner pointPartitioner(Global_nPointDomain,0);
 
-  /*--- Only call ParMETIS if we have more than one rank to avoid errors ---*/
-
-  if (size > SINGLE_NODE) {
-
-    /*--- Create some structures that ParMETIS needs for partitioning. ---*/
-
-    idx_t numflag, nparts, edgecut, wgtflag, ncon;
+  /*--- Some recommended defaults for the various ParMETIS options. ---*/
 
-    idx_t *vtxdist = new idx_t[size+1];
-    idx_t *part    = new idx_t[nPoint];
+  idx_t wgtflag = 2;
+  idx_t numflag = 0;
+  idx_t ncon    = 1;
+  real_t ubvec  = 1.0 + config->GetParMETIS_Tolerance();
+  idx_t nparts  = size;
+  idx_t options[METIS_NOPTIONS];
+  METIS_SetDefaultOptions(options);
+  options[1] = 0;
 
-    real_t ubvec;
-    real_t *tpwgts = new real_t[size];
+  /*--- Fill the necessary ParMETIS input data arrays. ---*/
 
-    /*--- Some recommended defaults for the various ParMETIS options. ---*/
+  vector<real_t> tpwgts(size, 1.0/size);
 
-    wgtflag = 0;
-    numflag = 0;
-    ncon    = 1;
-    ubvec   = 1.05;
-    nparts  = (idx_t)size;
-    idx_t options[METIS_NOPTIONS];
-    METIS_SetDefaultOptions(options);
-    options[1] = 0;
-
-    /*--- Fill the necessary ParMETIS data arrays. We do not apply
-     any weighting during the partitioning process. ---*/
+  vector<idx_t> vtxdist(size+1);
+  vtxdist[0] = 0;
+  for (int i = 0; i < size; i++) {
+    vtxdist[i+1] = pointPartitioner.GetLastIndexOnRank(i);
+  }
 
-    for (int i = 0; i < size; i++) {
-      tpwgts[i] = 1.0/((real_t)size);
-    }
+  /*--- For most FVM-type operations the amount of work is proportional to the
+   * number of edges, for a few however it is proportional to the number of points.
+   * Therefore, for (static) load balancing we consider a weighted function of points
+   * and number of edges (or neighbors) per point, giving more importance to the latter
+   * skews the partitioner towards evenly distributing the total number of edges. ---*/
 
-    vtxdist[0] = 0;
-    for (int i = 0; i < size; i++) {
-      vtxdist[i+1] = (idx_t)pointPartitioner.GetLastIndexOnRank(i);
-    }
+  const auto wp = config->GetParMETIS_PointWeight();
+  const auto we = config->GetParMETIS_EdgeWeight();
 
-    /*--- Calling ParMETIS ---*/
+  vector<idx_t> vwgt(nPoint);
+  for (unsigned long iPoint = 0; iPoint < nPoint; ++iPoint) {
+    vwgt[iPoint] = wp + we * (xadj[iPoint+1] - xadj[iPoint]);
+  }
```

Comment:
Weight function.

Reply:
If you have time, here's one suggestion to try to see if there's any benefit/difference.. ParMETIS is set up to handle multiple constraints per vertex. An alternative is to set ncon = 2 and then supply 2 contiguous weight values for each point. Something like:
```
  unsigned long idx = 0;
  for (unsigned long iPoint = 0; iPoint < nPoint; ++iPoint) {
     vwgt[idx++] = wp;
     vwgt[idx++] = we * (xadj[iPoint+1] - xadj[iPoint]);
   }
```